src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -198,20 +212,13 @@ public void visitLocalVariable(String name, String descriptor, String signature,
 
 			if (renameInvalidLocals && !isValidJavaIdentifier(name)) {
 				Type type = Type.getType(descriptor);
-				boolean plural = false;
+				name = getGeneratedName(type);
```

Comment:
It would be nice if this could be pulled from the appropriate parameter slot if it had already been visited (as that would also mean it was valid), although that would need all the `super.visitParameter` calls to have been saved to an array too to know.

Reply:
Oh, so you are saying, if visitParameter didn't find the name, just generate one there and then retrieve it here.
Yeah, that makes a lot of sense and actually also makes the access modifier thing a bit cleaner, too, since we can just always map at the end of visitParameter then.